socketio-agent/socketio-agent.js

```diff
@@ -19,7 +20,7 @@ module.exports = function(RED) {
         this.eventName = config.eventname;
         
         var server = RED.nodes.getNode(config.server);
-        this.socket = connect(server.uri);
+        this.socket = connect(server.uri, JSON.parse(server.options));
```

Comment:
Since the options are not required, potential user may set them to empty string. In that case, `JSON.parse()` will throw an error. Please change `JSON.parse(server.options)` to `JSON.parse(server.options || '{}')`. This will keep options input optional.